### PR TITLE
DEV-1061: ADR for DID Resolver

### PR DESCRIPTION
**Pull request by Toktar**

### CODE REVIEW

**Comment by ankurdotb** [comment]
Task linked: [DEV-1061 Define architecture for DID resolver (on-ledger) and universal resolver driver (off-ledger)](https://app.clickup.com/t/6600954/DEV-1061)

File: architecture/adr-list/adr-010-did-resolver.md
Comment (askolesov) [inline architecture/adr-list/adr-010-did-resolver.md:32]: Also,
* Convert arrays to maps somewhere (like in jwk_pubkey)
* In metadata, make `did` property to be of DID type instead of any

File: architecture/adr-list/adr-010-did-resolver.md
Comment (askolesov) [inline architecture/adr-list/adr-010-did-resolver.md:65]: Are those two compatible with each other?

File: architecture/adr-list/adr-010-did-resolver.md
Comment (Toktar) [reply architecture/adr-list/adr-010-did-resolver.md:32]: Not sure about the second one, but thank you! Will add them

File: architecture/adr-list/adr-010-did-resolver.md
Comment (Toktar) [reply architecture/adr-list/adr-010-did-resolver.md:65]: Sorry. Terrible wording. I meant synchronous answer for a client request

File: architecture/adr-list/adr-010-did-resolver.md
Comment (ankurdotb) [inline architecture/adr-list/adr-010-did-resolver.md:41]: Is this the raw draw.io XML file? I'd recommend instead of a link to embed a PNG version and separately I'll export the draw.io file itself for editing and link

**Review by ankurdotb — CHANGES_REQUESTED** [review]
Got a couple of comments as we flesh this out:

1. The bottom "consequences" section is intended in the ADR template for the final decision only, not for positive/negative per option. The two approaches should be described under the context section (you can mention pros/cons here).
2. I'd like you to go into a bit more detail on actual implementation as well. For example, with the selected approach for DID Resolver, how do we want to implement it: build from scratch or use a pre-existing library? We could either build from scratch in Go, or consider building using the DIF's TypeScript DID Resolver project: https://github.com/decentralized-identity/did-resolver
3. There's no discussion here on the Universal Resolver DID driver for cheqd. We should add this, by referencing the DIF Universal Resolver "Driver Development" guidance. https://github.com/decentralized-identity/universal-resolver/blob/main/docs/driver-development.md
4. The image/diagram right now is a blend of component as well as sequence/actor diagram, which is not ideal. Consider splitting them out. I'd like to see a sequence diagram for how the Universal Resolver + cheqd driver + DID resolver + ledger act together, done using swimlanes.io. (You can embed the PNG, and provide the swimlane text file as a link in case anyone wants to modify it.)

File: architecture/adr-list/adr-010-did-resolver.md
Comment (Toktar) [reply architecture/adr-list/adr-010-did-resolver.md:66]: From my point of view, it doesn't matter who is owner. Just if the web app is not available, the whole service not available

**Review by askolesov — COMMENTED** [review]
Thanks! The main question now - should we propose a library instead of `cheqd-did-service`?

File: architecture/adr-list/adr-010-did-resolver.md
Comment (Toktar) [reply architecture/adr-list/adr-010-did-resolver.md]: Done

File: architecture/adr-list/adr-010-did-resolver.md
Comment (Toktar) [reply architecture/adr-list/adr-010-did-resolver.md]: Done. Could you please check?

File: architecture/adr-list/adr-010-did-resolver.md
Comment (Toktar) [reply architecture/adr-list/adr-010-did-resolver.md]: Done

File: architecture/adr-list/adr-010-did-resolver.md
Comment (Toktar) [reply architecture/adr-list/adr-010-did-resolver.md]: Fixed

**Comment by Toktar** [comment]
ADR completed and approved by @ankurdotb .
**DO NOT MERGE** this PR until the Gitbook issues are fixed!

File: architecture/adr-list/adr-010-did-resolver.md
Comment (Toktar) [inline architecture/adr-list/adr-010-did-resolver.md:181]: @Tweeddalex 
"to setup" I believe?

File: architecture/adr-list/adr-010-did-resolver.md
Comment (Toktar) [inline architecture/adr-list/adr-010-did-resolver.md:189]: ...using Universal DID Resolver

**Review by Toktar — COMMENTED** [review]
Hey @Tweeddalex Thanks for your changes!
I added a few comments about your commit. Could you please take a look? I can fix them by myself if needed.
I pushed some small fixes for I believe just the typos.

File: architecture/adr-list/adr-010-did-resolver.md
Comment (Tweeddalex) [reply architecture/adr-list/adr-010-did-resolver.md:181]: Yep I agree, lets change to "to setup"

File: architecture/adr-list/adr-010-did-resolver.md
Comment (Tweeddalex) [reply architecture/adr-list/adr-010-did-resolver.md:189]: Sure, happy to change this

**Comment by Tweeddalex** [comment]
> @Tweeddalex @Toktar Has this been shifted over to the identity-docs repo?

Yes, I have added the latest version to the identity docs repo

**Pull request closed without merging**
